### PR TITLE
#4304 the stepwise plugin must be blocked on cacheprovider plugin block request

### LINKED ISSUE #4304: Possible bug with the 3.10.0 release
Hello,

Something went wrong with the 3.10.0 release 

```
   ============================= test session starts ==============================
    platform linux2 -- Python 2.7.12, pytest-3.10.0, py-1.7.0, pluggy-0.8.0
    rootdir: /builds/ansible-roles/json-inventory/molecule/default, inifile:
    plugins: testinfra-1.16.0
    collecting ... 
collected 3 items                                                              
    
Traceback (most recent call last):
  File "/usr/local/bin/py.test", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python2.7/dist-packages/_pytest/config/__init__.py", line 76, in main
    return config.hook.pytest_cmdline_main(config=config)
  File "/usr/local/lib/python2.7/dist-packages/pluggy/hooks.py", line 284, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
  File "/usr/local/lib/python2.7/dist-packages/pluggy/manager.py", line 67, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/usr/local/lib/python2.7/dist-packages/pluggy/manager.py", line 61, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
  File "/usr/local/lib/python2.7/dist-packages/pluggy/callers.py", line 208, in _multicall
    return outcome.get_result()
  File "/usr/local/lib/python2.7/dist-packages/pluggy/callers.py", line 81, in get_result
    _reraise(*ex)  # noqa
  File "/usr/local/lib/python2.7/dist-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
  File "/usr/local/lib/python2.7/dist-packages/_pytest/main.py", line 218, in pytest_cmdline_main
    return wrap_session(config, _main)
  File "/usr/local/lib/python2.7/dist-packages/_pytest/main.py", line 211, in wrap_session
    session=session, exitstatus=session.exitstatus
  File "/usr/local/lib/python2.7/dist-packages/pluggy/hooks.py", line 284, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
  File "/usr/local/lib/python2.7/dist-packages/pluggy/manager.py", line 67, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/usr/local/lib/python2.7/dist-packages/pluggy/manager.py", line 61, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
  File "/usr/local/lib/python2.7/dist-packages/pluggy/callers.py", line 203, in _multicall
    gen.send(outcome)
  File "/usr/local/lib/python2.7/dist-packages/_pytest/terminal.py", line 627, in pytest_sessionfinish
    outcome.get_result()
  File "/usr/local/lib/python2.7/dist-packages/pluggy/callers.py", line 81, in get_result
    _reraise(*ex)  # noqa
  File "/usr/local/lib/python2.7/dist-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
  File "/usr/local/lib/python2.7/dist-packages/_pytest/stepwise.py", line 102, in pytest_sessionfinish
    self.config.cache.set("cache/stepwise", [])
AttributeError: 'Config' object has no attribute 'cache'
```

The same nine hours ago with the 3.9.3 release :

```
    ============================= test session starts ==============================
    platform linux2 -- Python 2.7.12, pytest-3.9.3, py-1.7.0, pluggy-0.8.0
    rootdir: /builds/ansible-roles/json-inventory/molecule/default, inifile:
    plugins: testinfra-1.16.0
    
collecting 0 items                                                             
collecting 3 items                                                             
collected 3 items                                                              
    
    tests/test_default.py ...                                                [100%]
    
    =============================== warnings summary ===============================
    /usr/lib/python2.7/dist-packages/cryptography/x509/__init__.py:32
      /usr/lib/python2.7/dist-packages/cryptography/x509/__init__.py:32: PendingDeprecationWarning: CRLExtensionOID has been renamed to CRLEntryExtensionOID
        from cryptography.x509.oid import (
    
    -- Docs: https://docs.pytest.org/en/latest/warnings.html
    ==================== 3 passed, 1 warnings in 38.20 seconds =====================
``` 

If someone have an idea ? ;)

### PR DESCRIPTION
**Pull request by fzarifian**
Thanks for submitting a PR, your contribution is really appreciated!

Here's a quick checklist that should be present in PRs (you can delete this text from the final description, this is
just a guideline):

- [x] Create a new changelog file in the `changelog` folder, with a name like `<ISSUE NUMBER>.<TYPE>.rst`. See [changelog/README.rst](https://github.com/pytest-dev/pytest/blob/master/changelog/README.rst) for details.
- [x] Target the `master` branch for bug fixes, documentation updates and trivial changes.
- [ ] Target the `features` branch for new features and removals/deprecations.
- [ ] Include documentation when adding new features.
- [x] Include new tests or update existing tests when applicable.

Unless your change is trivial or a small documentation fix (e.g.,  a typo or reword of a small section) please:

- [x] Add yourself to `AUTHORS` in alphabetical order;

Fix #4304

### CODE REVIEW

File: src/_pytest/config/__init__.py [hunk 1]
```diff
@@ -476,6 +476,9 @@ def consider_preparse(self, args):
     def consider_pluginarg(self, arg):
         if arg.startswith("no:"):
             name = arg[3:]
+            # PR #4304 : remove stepwise if cacheprovider is blocked
+            if name == "cacheprovider":
+                self.consider_pluginarg("no:stepwise")
```
Comment (RonnyPfannschmidt) [inline src/_pytest/config/__init__.py]: please use `self.set_blocked("stepwhise")`

**Comment by nicoddemus** [comment]
Thanks @fzarifian! Would you like to add yourself to `AUTHORS`?

**Comment by nicoddemus** [comment]
Hmm perhaps we should follow the general idea taken in https://github.com/pytest-dev/pytest/pull/4335 and instead set `active` to False in the stepwise plugin if `config` doesn't have a `cache` attribute?

**Comment by nicoddemus** [comment]
Hmm but I see this has the advantage of working with the standalone `pytest-stepwise` plugin too, although the current released version does not work with the latest pytest anyway.

**Comment by fzarifian** [comment]
Thanks a lot for your help resolving the issue ;)
Have a nice day

**Comment by nicoddemus** [comment]
In the end I don't think it matters much, merging. Thanks @fzarifian again!

**Pull request merged**